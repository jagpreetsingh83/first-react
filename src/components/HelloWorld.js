import React, {Component} from 'react';
import {Button} from 'react-bootstrap';
import './HelloWorld.css';

class HelloWorld extends Component {
    constructor(props) {
        super(props);
        this.state = {
            greeting: 'Hello'
        }
        this.frenchify = this
            .frenchify
            .bind(this);
        this.deleteGreeting = this
            .deleteGreeting
            .bind(this);
    }
    frenchify() {
        this.setState({greeting: 'Bonjour'});
    }
    deleteGreeting(e) {
        this
            .props
            .removeGreeting(this.props.name);
    }
    render() {
        return (
            <div className="HelloWorld">{this.state.greeting} {this.props.name}!
                <Button
                    onClick={this.deleteGreeting}
                    bsStyle="danger"
                    bsSize="xsmall"
                    className="pull-right">Delete</Button>
                <Button
                    onClick={this.frenchify}
                    bsStyle="primary"
                    bsSize="xsmall"
                    className="pull-right">Frenchify!</Button>
            </div>
        );
    }
}

export default HelloWorld;
