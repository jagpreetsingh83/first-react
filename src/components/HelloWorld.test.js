import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import HelloWorld from './HelloWorld';

describe(HelloWorld, () => {
    /*
    I usually like to tackle my React tests in the following way
    1.  Write a generic snapshot test
    2.  Write some sub-component-specific tests
    3.  Check content
    4.  Check interactions. Given that, letâ€™s start tackling these in order.
    */
    const name = 'Person';
    const mockRemoveGreeting = jest.fn();
    const component = shallow(<HelloWorld name={name} removeGreeting={mockRemoveGreeting}/>);

    // Writing A Generic Snapshot Test
    it('renders and matches our snapshot', () => {
        const component = renderer.create(<HelloWorld name="Person"/>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    // Writing Content Tests
    it('contains the supplied name', () => {
        expect(component.text()).toContain(name);
    });
    // Checking Interactions
    it('modifies the greeting when frenchify button is clicked', () => {
        component
            .find('Button.french')
            .simulate('click');
        expect(component.text()).toContain('Bonjour');
    });
    it("calls the passed in removeGreeting function when remove button is clicked", () => {
        component
            .find("Button.delete")
            .simulate("click");
        expect(mockRemoveGreeting).toBeCalled();
    });
});