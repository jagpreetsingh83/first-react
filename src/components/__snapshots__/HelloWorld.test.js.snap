// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class HelloWorld extends _react.Component {
    constructor(props) {
        super(props);
        this.state = {
            greeting: 'Hello' };

        this.frenchify = this.
        frenchify.
        bind(this);
        this.deleteGreeting = this.
        deleteGreeting.
        bind(this);
    }
    frenchify() {
        this.setState({ greeting: 'Bonjour' });
    }
    deleteGreeting(e) {
        this.
        props.
        removeGreeting(this.props.name);
    }
    render() {
        return (
            _react2.default.createElement('div', { className: 'HelloWorld', __source: { fileName: _jsxFileName, lineNumber: 28 }, __self: this }, this.state.greeting, ' ', this.props.name, '!',
                _react2.default.createElement(_reactBootstrap.Button, {
                        onClick: this.deleteGreeting,
                        bsStyle: 'danger',
                        bsSize: 'xsmall',
                        className: 'pull-right delete', __source: { fileName: _jsxFileName, lineNumber: 29 }, __self: this }, 'Delete'),
                _react2.default.createElement(_reactBootstrap.Button, {
                        onClick: this.frenchify,
                        bsStyle: 'primary',
                        bsSize: 'xsmall',
                        className: 'pull-right french', __source: { fileName: _jsxFileName, lineNumber: 34 }, __self: this }, 'Frenchify!')));


    }} renders and matches our snapshot 1`] = `
<div
  className="HelloWorld"
>
  Hello
   
  Person
  !
  <button
    className="pull-right delete btn btn-xs btn-danger"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Delete
  </button>
  <button
    className="pull-right french btn btn-xs btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Frenchify!
  </button>
</div>
`;
